tools/run_tests/xds_k8s_test_driver/tests/url_map/metadata_filter_test.py

```diff
@@ -0,0 +1,270 @@
+        self.assertEqual(len(xds_config.rds['virtualHosts'][0]['routes']), 2)
+        self.assertEqual(
+            xds_config.rds['virtualHosts'][0]['routes'][0]['match']['prefix'],
+            "/grpc.testing.TestService/Unary")
+        self.assertEqual(
+            xds_config.rds['virtualHosts'][0]['routes'][1]['match']['prefix'],
+            "")
+        logger.info('config:%s', xds_config)
+
+    def rpc_distribution_validate(self, test_client: XdsTestClient):
+        rpc_distribution = self.configure_and_send(test_client,
+                                                   rpc_types=[RpcTypeUnaryCall],
+                                                   num_rpcs=_NUM_RPCS)
+        self.assertEqual(
+            _NUM_RPCS,
+            rpc_distribution.unary_call_alternative_service_rpc_count)
+
+
+class TestMetadataFilterMatchMultipleRules(
+        xds_url_map_testcase.XdsUrlMapTestCase):
+
+    @staticmethod
+    def url_map_change(
+            host_rule: HostRule,
+            path_matcher: PathMatcher) -> Tuple[HostRule, PathMatcher]:
+        path_matcher["routeRules"] = [{
+            'priority': 0,
+            'matchRules': [{
+                'prefixMatch':
+                    '/',
+                'headerMatches': [{
+                    'headerName': _TEST_METADATA_KEY,
+                    'exactMatch': _TEST_METADATA_VALUE_EMPTY
+                }],
+                'metadataFilters': [{
+                    'filterMatchCriteria': 'MATCH_ANY',
+                    'filterLabels': match_labels
+                }]
+            }],
+            'service': GcpResourceManager().alternative_backend_service()
+        }, {
+            'priority': 1,
+            'matchRules': [{
+                'prefixMatch':
+                    '/grpc.testing.TestService/Unary',
+                'metadataFilters': [{
+                    'filterMatchCriteria': 'MATCH_ALL',
+                    'filterLabels': match_labels
+                }]
+            }],
+            'service': GcpResourceManager().default_backend_service()
+        }]
+        return host_rule, path_matcher
+
+    def xds_config_validate(self, xds_config: DumpedXdsConfig):
+        self.assertNumEndpoints(xds_config, 2)
+        self.assertEqual(len(xds_config.rds['virtualHosts'][0]['routes']), 3)
+        self.assertEqual(
+            xds_config.rds['virtualHosts'][0]['routes'][0]['match']['headers']
+            [0]['name'], _TEST_METADATA_KEY)
+        self.assertEqual(
+            xds_config.rds['virtualHosts'][0]['routes'][0]['match']['headers']
+            [0]['exactMatch'], _TEST_METADATA_VALUE_EMPTY)
+        self.assertEqual(
+            xds_config.rds['virtualHosts'][0]['routes'][1]['match']['prefix'],
+            "/grpc.testing.TestService/Unary")
+        self.assertEqual(
+            xds_config.rds['virtualHosts'][0]['routes'][2]['match']['prefix'],
+            "")
+        logger.info('config:%s', xds_config)
```

Comment:
The config will be dumped if this test case failed. If we log at every attempt, the driver log can be quite verbose.